### PR TITLE
Upgrade some golang.org/x libraries to get CVE patches

### PR DESCRIPTION
**Pull request by denkyl08**
Should address [CVE-2021-38561](https://osv.dev/vulnerability/GO-2021-0113), [CVE-2021-33194](https://security.archlinux.org/CVE-2021-33194), and CVE-2020-29652

### CODE REVIEW

**Comment by denkyl08** [comment]
Updated to address CVEs in client-go package. This also addresses https://github.com/mailgun/gubernator/issues/53. That log message wasn't a functional issue. I believe that these messages occur when etcd compaction occurs and the resourceVersion used to initialize the watch is no longer in storage.  See https://github.com/kubernetes/kubernetes/issues/22024. This log message was changed from a warning to info here: https://github.com/kubernetes/client-go/commit/23e6ea699ebd2d6b88dd81454aea527dbc3a77ec but will not be logged at all if log verbosity is <4.

File: kubernetes.go [hunk 1]
```diff
@@ -82,11 +85,15 @@ func NewK8sPool(conf K8sPoolConfig) (*K8sPool, error) {
...
-		done:   make(chan struct{}),
-		log:    conf.Logger,
-		client: client,
-		conf:   conf,
+		done:        make(chan struct{}),
+		log:         conf.Logger,
+		client:      client,
+		conf:        conf,
+		watchCtx:    ctx,
+		cancelWatch: fn,
```
Comment (Baliedge) [inline kubernetes.go:96]: To be consistent, I'd like to see `watch` as the prefix for both these new fields.

**Review by Baliedge — CHANGES_REQUESTED** [review]
Small nomenclature changes requested.

**Review by Baliedge — APPROVED** [review]
👍🏻

**Pull request merged**